src/coreclr/vm/prestub.cpp

```diff
@@ -1059,7 +1045,431 @@ PCODE MethodDesc::JitCompileCodeLocked(PrepareCodeConfig* pConfig, JitListLockEn
+
+        // Check the name of the target to access. This is the name we
+        // use to look up the intended token in metadata.
+        CaNamedArg namedArgs[1];
+        CaType namedArgTypes[1];
+        namedArgTypes[0].Init(SERIALIZATION_TYPE_STRING);
+        namedArgs[0].Init("Name", SERIALIZATION_TYPE_PROPERTY, namedArgTypes[0], NULL);
+        if (FAILED(::ParseKnownCaNamedArgs(ca, namedArgs, ARRAY_SIZE(namedArgs))))
+            return false;
+
+        // If the Name isn't defined, then use the name of the method.
+        if (namedArgs[0].val.type.tag == SERIALIZATION_TYPE_UNDEFINED)
+        {
+            // The Constructor case has an implied value provided by
+            // the runtime. We are going to enforce this during consumption
+            // so we avoid the setting of the value. We validate the name
+            // as empty at the use site.
+            if (kind != UnsafeAccessorKind::Constructor)
+                name.SetUTF8(pMD->GetName());
+        }
+        else
+        {
+            const CaValue& val = namedArgs[0].val;
+            name.SetUTF8(val.str.pStr, val.str.cbStr);
+        }
+
+        return true;
+    }
+
+    struct GenerationContext final
+    {
+        GenerationContext(UnsafeAccessorKind kind, MethodDesc* pMD)
+            : Kind{ kind }
+            , Declaration{ pMD }
+            , DeclarationSig{ pMD }
+            , TargetType{}
+            , IsTargetStatic{ false }
+            , TargetMethod{}
+            , TargetField{}
+        { }
+
+        UnsafeAccessorKind Kind;
+        MethodDesc* Declaration;
+        MetaSig DeclarationSig;
+        TypeHandle TargetType;
+        bool IsTargetStatic;
+        MethodDesc* TargetMethod;
+        FieldDesc* TargetField;
+    };
+
+    bool TrySetTargetMethod(
+        GenerationContext& cxt,
+        LPCUTF8 methodName)
+    {
+        STANDARD_VM_CONTRACT;
+        _ASSERTE(methodName != NULL);
+        _ASSERTE(cxt.Kind == UnsafeAccessorKind::Constructor
+                || cxt.Kind == UnsafeAccessorKind::Method
+                || cxt.Kind == UnsafeAccessorKind::StaticMethod);
+
+        TypeHandle targetType = cxt.TargetType.IsByRef()
+            ? cxt.TargetType.GetTypeParam()
+            : cxt.TargetType;
+
+        // Due to how some types degrade, unsafe access on these
+        // target types is blocked, even if the lookup succeeds.
+        // For example, pointers can become lookups on typeof(uint)'s MethodTable.
+        if (targetType.IsTypeDesc())
+            ThrowHR(COR_E_BADIMAGEFORMAT, BFA_INVALID_UNSAFEACCESSOR);
+
+        CorElementType declCorType;
+        CorElementType maybeCorType;
+        TypeHandle declType;
+        TypeHandle maybeType;
+
+        // Following the iteration pattern found in MemberLoader::FindMethod().
+        // Reverse order is recommended - see comments in MemberLoader::FindMethod().
+        MethodTable::MethodIterator iter(targetType.GetMethodTable());
```

Comment:
```suggestion
        MethodTable::MethodIterator iter(targetType.AsMethodTable());
```
Nit: We have rejected the typedesc above. So this can use `AsMethodTable` that is a bit cheaper.